docs/DevGuide/Variables.md

```diff
@@ -0,0 +1,75 @@
+\cond NEVER
```

Comment:
You probably want this to be visible in `docs/DevGuide/DevGuide.md`. Maybe under the `Foundational Concepts in SpECTRE` add a new entry in the list:
```
- \subpage variables_foundations "Using the Variables class to improve efficiency"
```
Wording is up to you :)